contrib/gin-gonic/gin/gintrace.go

```diff
@@ -34,27 +33,21 @@ func Middleware(service string, opts ...Option) gin.HandlerFunc {
 		if cfg.ignoreRequest(c) {
 			return
 		}
-		resource := cfg.resourceNamer(c)
 		opts := []ddtrace.StartSpanOption{
-			tracer.ServiceName(cfg.serviceName),
-			tracer.ResourceName(resource),
-			tracer.SpanType(ext.SpanTypeWeb),
-			tracer.Tag(ext.HTTPMethod, c.Request.Method),
-			tracer.Tag(ext.HTTPURL, c.Request.URL.Path),
 			tracer.Measured(),
 		}
 		if !math.IsNaN(cfg.analyticsRate) {
 			opts = append(opts, tracer.Tag(ext.EventSampleRate, cfg.analyticsRate))
 		}
-		if spanctx, err := tracer.Extract(tracer.HTTPHeadersCarrier(c.Request.Header)); err == nil {
-			opts = append(opts, tracer.ChildOf(spanctx))
-		}
-		span, ctx := tracer.StartSpanFromContext(c.Request.Context(), "http.request", opts...)
-		defer span.Finish()
+		span, ctx := httptrace.StartRequestSpan(c.Request, cfg.serviceName, cfg.resourceNamer(c), false, opts...)
 
 		// pass the span through the request context
 		c.Request = c.Request.WithContext(ctx)
 
+		defer func() {
+			httptrace.FinishRequestSpan(span, c.Writer.Status())
+		}()
```

Comment:
```suggestion
		defer httptrace.FinishRequestSpan(span, c.Writer.Status())
```
Same question here. Also, why not put the defer just below the `StartRequestSpan` ? Is there anything important that `c.Request.WithContext(ctx)` does which changes the output of `c.Writer().Status()`?

If possible, it's nice to group the `defer` together with the thing it relates to.

Reply:
The only reason why I've moved this right after the span creation was to avoid the common mistake of using the request context right after like if it was replaced by the new one while it wasn't yet.

This code doesn't do that so I'm good to move it back below defer everywhere I've done that. Let me know.